### PR TITLE
[luci] Revise vector access implementation in sparsifier

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit revises vector access implentation in sparsifier and add
related test.

from [] operator to .at method for preventing from undefined behavior.

Related : #3889
Draft: #4394 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
Can you add some negative tests ?

**Comment by mhs4670go** [comment]
@seanshpark PTAL:)

File: compiler/luci/pass/src/Sparsifier.test.cpp [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  EXPECT_EQ(expected_dm6, dim_metadata[6]);
+  EXPECT_EQ(expected_dm1, dim_metadata[7]);
+
+  const auto data = sparsifier.GetData();
+  const std::vector<int32_t> expected_data = {0,  1,  2,  3,  4,  5,  6,  7,  8,  9,
+                                              10, 11, 12, 13, 14, 15, 20, 21, 22, 23};
+  EXPECT_EQ(expected_data, data);
+}
+
+TEST(SparsifierTest, WrongTraversalOrderRank_neg)
```
Comment (seanshpark) [inline compiler/luci/pass/src/Sparsifier.test.cpp:169]: please use uppercase `_NEG` :)

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**